crates/unleash-edge/src/edge_builder.rs

```diff
@@ -1,24 +1,36 @@
-use crate::{CacheContainer, EdgeInfo};
+use crate::{CacheContainer, EdgeInfo, SHOULD_DEFER_VALIDATION};
 use chrono::Duration;
 use dashmap::DashMap;
+use std::pin::Pin;
 use std::sync::Arc;
+use tokio::sync::RwLock;
 use tokio::sync::mpsc::UnboundedSender;
-use tracing::{debug, error, warn};
-use unleash_edge_auth::token_validator::TokenValidator;
-use unleash_edge_cli::{AuthHeaders, EdgeArgs, RedisMode};
+use tracing::{debug, error, info, warn};
+use unleash_edge_appstate::AppState;
+use unleash_edge_auth::token_validator::{
+    TokenValidator, create_revalidation_of_startup_tokens_task, create_revalidation_task,
+};
+use unleash_edge_cli::{AuthHeaders, CliArgs, EdgeArgs, RedisMode};
 use unleash_edge_delta::cache_manager::DeltaCacheManager;
 use unleash_edge_feature_cache::FeatureCache;
 use unleash_edge_feature_refresh::{FeatureRefreshConfig, FeatureRefresher};
-use unleash_edge_http_client::{ClientMetaInformation, UnleashClient};
-use unleash_edge_persistence::EdgePersistence;
+use unleash_edge_http_client::instance_data::{
+    InstanceDataSending, create_send_instance_data_task,
+};
+use unleash_edge_http_client::{
+    ClientMetaInformation, HttpClientArgs, UnleashClient, new_reqwest_client,
+};
+use unleash_edge_metrics::axum_prometheus_metrics::PrometheusAxumLayer;
+use unleash_edge_metrics::metrics_pusher::create_prometheus_write_task;
+use unleash_edge_metrics::send_unleash_metrics::create_send_metrics_task;
 use unleash_edge_persistence::file::FilePersister;
 use unleash_edge_persistence::redis::RedisPersister;
-
-#[cfg(feature = "s3-persistence")]
```

Comment:
Seems like we lost this along the way. Is it intentional?

Reply:
Nope, that was my bad. It does seem to compile without + no-default-features it but I think it's better to have it in. Fixed now